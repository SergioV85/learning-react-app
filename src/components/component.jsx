import React from 'react';

export class Subtitle extends React.Component {
  render() {
    return <h2>This is subtitle, simple component</h2>;
  }
}