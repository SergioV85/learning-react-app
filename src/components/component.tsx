import * as React from 'react';

export class Subtitle extends React.Component {
  public render() {
    return <h2 className="custom-header">This is subtitle, simple component</h2>;
  }
}