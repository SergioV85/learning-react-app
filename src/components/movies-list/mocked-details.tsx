export const mockedMovieDetails = {
  "budget": 0,
  "id": 419831,
  "overview": "Barbara Thorson struggles through life by escaping into a fantasy life of magic and monsters.",
  "poster_path": "https://image.tmdb.org/t/p/w500/cvit6HDbXHE6W5kGPd47jd0wthQ.jpg",
  "release_date": "2018-01-21",
  "revenue": 0,
  "runtime": 106,
  "tagline": "",
  "title": "I Kill Giants",
  "vote_average": 6,
  "vote_count": 63,
  "genres": [
    "Thriller",
    "Drama",
    "Fantasy"
  ],
}