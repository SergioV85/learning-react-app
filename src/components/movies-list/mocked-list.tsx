export const mockedMoviesList = [{
    "id": 419831,
    "title": "I Kill Giants",
    "tagline": "",
    "vote_average": 6,
    "vote_count": 63,
    "release_date": "2018-01-21",
    "poster_path": "https://image.tmdb.org/t/p/w500/cvit6HDbXHE6W5kGPd47jd0wthQ.jpg",
    "overview": "Barbara Thorson struggles through life by escaping into a fantasy life of magic and monsters.",
    "budget": 0,
    "revenue": 0,
    "genres": [
      "Thriller",
      "Drama",
      "Fantasy"
    ],
    "runtime": 106
  },
  {
    "id": 399057,
    "title": "The Killing of a Sacred Deer",
    "tagline": "",
    "vote_average": 6.6,
    "vote_count": 509,
    "release_date": "2017-10-20",
    "poster_path": "https://image.tmdb.org/t/p/w500/e4DGlsc9g0h5AyoyvvAuIRnofN7.jpg",
    "overview": "A teenager's attempts to bring a brilliant surgeon into his dysfunctional family take an unexpected turn.",
    "budget": 0,
    "revenue": 2205074,
    "genres": [
      "Drama",
      "Horror",
      "Thriller"
    ],
    "runtime": 121
  },
  {
    "id": 423988,
    "title": "Kill Switch",
    "tagline": "Worlds will collide",
    "vote_average": 5,
    "vote_count": 81,
    "release_date": "2017-06-01",
    "poster_path": "https://image.tmdb.org/t/p/w500/8K6pfMcT3hiIlCFw2RrHoXLshxl.jpg",
    "overview": "A pilot battles to save his family and the planet after an experiment for unlimited energy goes wrong.",
    "budget": 1000000,
    "revenue": 0,
    "genres": [
      "Science Fiction"
    ],
    "runtime": 91
  },
  {
    "id": 382322,
    "title": "Batman: The Killing Joke",
    "tagline": "The madness begins.",
    "vote_average": 6.2,
    "vote_count": 618,
    "release_date": "2016-07-21",
    "poster_path": "https://image.tmdb.org/t/p/w500/zm0ODjtfJfJW0W269LqsQl5OhJ8.jpg",
    "overview": "As Batman hunts for the escaped Joker, the Clown Prince of Crime attacks the Gordon family to prove a diabolical point mirroring his own fall into madness. Based on the graphic novel by Alan Moore and Brian Bolland.",
    "budget": 3500000,
    "revenue": 3775000,
    "genres": [
      "Action",
      "Animation",
      "Crime",
      "Drama"
    ],
    "runtime": 72
  },
  {
    "id": 237756,
    "title": "Kill Me Three Times",
    "tagline": "Once is never enough.",
    "vote_average": 5.5,
    "vote_count": 203,
    "release_date": "2015-04-10",
    "poster_path": "https://image.tmdb.org/t/p/w500/zqTlUUBlVZTzVMYUnJQve0kJc3o.jpg",
    "overview": "While on a seemingly routine job, a jaded hit man discovers that he's not the only one with his target in the crosshairs.",
    "budget": 0,
    "revenue": 0,
    "genres": [
      "Comedy",
      "Thriller"
    ],
    "runtime": 90
  }
]