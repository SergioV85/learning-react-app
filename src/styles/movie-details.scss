.c-movie-details {
  display: flex;
  flex-direction: row;
  margin: 30px 0;
  width: 100%;
  
  &__poster {
    margin-right: 20px;
    max-height: 250px;
    width: auto;
  }
  &__info {
    flex: 1 1 auto;
  }
  &__header {
    display: flex;
  }
  &__title {
    color: $light-red;
    font-family: $tilte-font;
    font-size: 1.2rem;
    font-weight: 700;
    margin-right: 10px;
  }
  &__rating {
    align-items: center;
    border: 1px solid $primary-green;
    border-radius: 100%;
    color: $light-green;
    display: flex;
    font-family: $tilte-font;
    font-size: 0.9rem;
    height: 24px;
    justify-content: center;
    width: 24px;
  }
  &__tagline {
    color: $primary-grey;
    font-family: $main-font;
    font-size: 0.9rem;
  }
  &__year {
    color: $primary-light;
    font-family: $main-font;
    font-size: 1rem;
    margin-right: 20px;
  }
  &__duration {
    color: $primary-light;
    font-family: $main-font;
    font-size: 1rem;
  }
  &__description {
    color: $primary-grey;
    font-family: $main-font;
    font-size: 0.9rem;
    margin-top: 20px;
  }
}