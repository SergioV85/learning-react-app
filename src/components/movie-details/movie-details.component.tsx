import * as React from 'react';
import { DateTime } from 'luxon';
import { MovieObject } from './../movie-card/movie-card.component';

interface MovieDetailsProps {
  selectedMovie: MovieObject
}

interface MovieDetailsState {}

export class MovieDetails extends React.Component<MovieDetailsProps, MovieDetailsState> {
  constructor(props: MovieDetailsProps) {
    super(props);
  }

  public render() {
    return <div className="c-movie-details">
      <img className="c-movie-details__poster" src={this.props.selectedMovie.poster_path} />
      <div className="c-movie-details__info">
        <h4 className="c-movie-details__header">
          <span className="c-movie-details__title">{this.props.selectedMovie.title}</span>
          <span className="c-movie-details__rating">{this.props.selectedMovie.vote_average}</span>
        </h4>
        <p className="c-movie-details__tagline">
          {this.props.selectedMovie.tagline}
        </p>
        <p>
          <span className="c-movie-details__year">
            { DateTime.fromISO(this.props.selectedMovie.release_date).year }
          </span>
          <span className="c-movie-details__duration">
            {this.props.selectedMovie.runtime} min
          </span>
        </p>
        <div className="c-movie-details__description">
          {this.props.selectedMovie.overview}
        </div>
      </div>
    </div>;
  }
}