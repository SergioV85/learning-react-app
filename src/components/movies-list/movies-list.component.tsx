import * as React from 'react';
import { ErrorBoundary } from './../error-boundary/error-boundary.component';
import { MovieCard, MovieObject } from './../movie-card/movie-card.component';
interface MoviesListProps {
  movies: MovieObject[];
  onMovieClick(): void;
}

interface MoviesListState {}

export class MoviesList extends React.Component<MoviesListProps, MoviesListState> {
  constructor(props: MoviesListProps) {
    super(props);
  }

  public render() {
    return <div className={`c-movies-list ${this.props.movies.length === 0 ? 'c-movies-list--empty' : ''}`}>
      {
        this.props.movies.length === 0
        ? <h1 className="c-movies-list__no-found-title">No films found</h1>
        : <ul className="c-movies-list__collection container">
            {this.props.movies.map(
              (movie) => 
                <ErrorBoundary key={`movie-card-wrapper-${movie.id}`} errorMessage={`Unfortunally, we can't show ${movie.title} preview`}>
                  <MovieCard key={movie.id} movie={movie} onMovieClick={this.props.onMovieClick} />
                </ErrorBoundary>
            )}
          </ul>
      }
    </div>;
  }
}