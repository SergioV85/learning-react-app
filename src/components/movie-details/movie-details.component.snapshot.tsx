import * as React from 'react';
import { shallow } from 'enzyme';
import { MovieDetails } from './movie-details.component';
import { mockedMovieDetails } from './../movies-list/mocked-details';

describe('Movie Details Component Snapshot', () => {
  it('should renders correctly', () => {
    const movieDetailsSnapshot = shallow(
      <MovieDetails selectedMovie={mockedMovieDetails} />
    );
    expect(movieDetailsSnapshot)
      .toMatchSnapshot();
  });
});