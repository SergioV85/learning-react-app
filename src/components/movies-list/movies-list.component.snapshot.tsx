import * as React from 'react';
import { shallow } from 'enzyme';
import { MoviesList } from './movies-list.component';
import { mockedMoviesList } from './mocked-list';

describe('Movies List Component Snapshot', () => {
  it('should renders correctly', () => {
    const moviesListSnapshot = shallow(
      <MoviesList movies={mockedMoviesList} onMovieClick={() => {}} />
    );
    expect(moviesListSnapshot)
      .toMatchSnapshot();
  });
});