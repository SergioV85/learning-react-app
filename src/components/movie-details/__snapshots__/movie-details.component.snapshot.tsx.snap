// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Details Component Snapshot should renders correctly 1`] = `
<div
  className="c-movie-details"
>
  <img
    className="c-movie-details__poster"
    src="https://image.tmdb.org/t/p/w500/cvit6HDbXHE6W5kGPd47jd0wthQ.jpg"
  />
  <div
    className="c-movie-details__info"
  >
    <h4
      className="c-movie-details__header"
    >
      <span
        className="c-movie-details__title"
      >
        I Kill Giants
      </span>
      <span
        className="c-movie-details__rating"
      >
        6
      </span>
    </h4>
    <p
      className="c-movie-details__tagline"
    />
    <p>
      <span
        className="c-movie-details__year"
      >
        2018
      </span>
      <span
        className="c-movie-details__duration"
      >
        106
         min
      </span>
    </p>
    <div
      className="c-movie-details__description"
    >
      Barbara Thorson struggles through life by escaping into a fantasy life of magic and monsters.
    </div>
  </div>
</div>
`;
